pulumitest/pulumiTest.go

```diff
@@ -39,6 +43,39 @@ func NewPulumiTest(t PT, source string, opts ...opttest.Option) *PulumiTest {
 	return pt
 }
 
+// NewPulumiYamlTest creates a new PulumiYamlTest instance.
+func NewInlinePulumiTest(t PT, program *pulumiyaml.Program, opts ...opttest.Option) *PulumiTest {
+	t.Helper()
+	ctx := testContext(t)
+	options := opttest.DefaultOptions()
+	for _, opt := range opts {
+		opt.Apply(options)
+	}
+	if program.Runtime == "" {
+		program.Runtime = "yaml"
+	}
+	if program.Name == "" {
+		program.Name = "test"
+	}
+	workingDir := tempDirWithoutCleanupOnFailedTest(t, "yamlProgramDir")
```

Comment:
If the runtime can be set to anything, then it shouldn't be called `"yamlProgramDir"`. If this is only for YAML, then we should assert that `program.Runtime == "yaml" || program.Runtime == ""`.